Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+﻿using RimWorld;
+using SaveOurShip2;
+using System.Text;
+using Verse;
+
+namespace TheCafFiend
+{
+    [StaticConstructorOnStartup]
+    public class CompSpinalEngineTrail : CompEngineTrail
+    {
+        public override string CompInspectStringExtra()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (fullyFormed)
+            {
+                stringBuilder.Append($"Fuel burn rate per second: {cachedFuelUse} of {cachedFuelAllowed} supported");
+                stringBuilder.AppendInNewLine($"Thrust: {cachedThrust * 500}");
+            }
+            else
+            {
+                stringBuilder.Append(cachedError.Colorize(UnityEngine.Color.red));
+            }
+            return stringBuilder.ToString();
+        }
+        public virtual CompProperties_SpinalEngineTrail Properties //Wait is this dirty I just want to be clear it's not hiding hrm
```

Comment:
I don't think? This is a case of using new because it's a different name, no? I was not very sure of the better approach: I wanted something that was explicitly only ever used by my own code, on these objects, which if I understood your handy new vs overide article fully (press X to doubt)  would be best as an override (were it the same name, which maybe it should be?) because if something within compEngineTrail called Props on a CompSpinalEngineTrail then with a new keyword to hide the base, it would call the non-spinal version? (The convertible example). I went with a different name partly because I was sick of trying to keep all the Props props Props straight in my head but I'm not entirely married to that being the correct answer either tbh